### PR TITLE
changed the structure of the contrib/python folder

### PR DESCRIPTION
**Pull request by mgouiza**
I suggest to change the structure of the contrib/python folder by adding 2 subfolders:
- scripts: for useful python functions and scripts
- notebooks: for notebooks
I also made some minor changes to the existing notebook to account for these changes.

I also updated the readme.md file

Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

*Describe what you did in this PR and why you did it.*

### Before your first pull request:

* [ ] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [ ] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [ ] I have tested my new feature locally to ensure it is correct.
* [ ] I have [created a testcase](https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [ ] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: contrib/python/notebooks/read_output_files/example.ipynb
Comment (gassmoeller) [inline contrib/python/notebooks/read_output_files/example.ipynb:1]: Remove the output that is stored in this Jupyter notebook.

File: contrib/python/scripts/__pycache__/aspect_data.cpython-310.pyc
Comment (gassmoeller) [inline contrib/python/scripts/__pycache__/aspect_data.cpython-310.pyc:1]: Remove this cache file

File: contrib/python/README.md [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+- numpy
+- io 
+- os
+- subprocess
+- sys 
+- pandas
+- xmltodict
+- requests
+- matplotlib
+- pyvista
```
Comment (gassmoeller) [inline contrib/python/README.md:16]: Please add an environment.yml file like for the documentation (see /doc/sphinx/environment.yml). This way one can set up the conda environment as `conda env create -f environment.yml`.

File: contrib/python/README.md [hunk 2]
```diff
@@ -1,14 +1,25 @@
 Python contributions for ASPECT 
 ==============================
 
-This folder contains some useful routines to read files with Python,
-using the numpy and pandas packages.
+This folder contains python function that ASPECT users can find useful.
```
Comment (gassmoeller) [inline contrib/python/README.md:4]:
```suggestion
This folder contains python functions and notebooks that ASPECT users can find useful.
```

**Review by gassmoeller — COMMENTED** [review]
Close, just some cleanup comments.

File: contrib/python/README.md (hunk 1 above)
Comment (mgouiza) [reply contrib/python/README.md]: Done

File: contrib/python/scripts/__pycache__/aspect_data.cpython-310.pyc
Comment (mgouiza) [reply contrib/python/scripts/__pycache__/aspect_data.cpython-310.pyc:1]: done

File: contrib/python/notebooks/read_output_files/example.ipynb
Comment (mgouiza) [reply contrib/python/notebooks/read_output_files/example.ipynb:1]: done

**Review by gassmoeller — APPROVED** [review]
Somehow this got overlooked during the last days of the hackathon. Ready to merge.

**Pull request merged**